sympy/core/equation.py

```diff
@@ -0,0 +1,508 @@
+                            check=False)
+        elif isinstance(a, Equation) and isinstance(b, Equation):
+            return Equation(opfunc_ab(a.lhs, b.lhs), opfunc_ab(a.rhs, b.rhs),
+                            check=False)
+        else:
+            raise TypeError('One of a or b should be an equation')
+
+    def __add__(self, other):
+        return self._binary_op(self, other, lambda a, b: a + b)
+
+    def __radd__(self, other):
+        return self._binary_op(other, self, lambda a, b: a + b)
+
+    def __mul__(self, other):
+        return self._binary_op(self, other, lambda a, b: a * b)
+
+    def __rmul__(self, other):
+        return self._binary_op(other, self, lambda a, b: a * b)
+
+    def __sub__(self, other):
+        return self._binary_op(self, other, lambda a, b: a - b)
+
+    def __rsub__(self, other):
+        return self._binary_op(other, self, lambda a, b: a - b)
+
+    def __truediv__(self, other):
+        return self._binary_op(self, other, lambda a, b: a / b)
+
+    def __rtruediv__(self, other):
+        return self._binary_op(other, self, lambda a, b: a / b)
+
+    def __mod__(self, other):
+        return self._binary_op(self, other, lambda a, b: a % b)
+
+    def __rmod__(self, other):
+        return self._binary_op(other, self, lambda a, b: a % b)
+
+    def __pow__(self, other):
+        return self._binary_op(self, other, lambda a, b: a ** b)
+
+    def __rpow__(self, other):
+        return self._binary_op(other, self, lambda a, b: a ** b)
+
+    def _eval_power(self, other):
+        return self.__pow__(other)
+
+    #####
+    # Operation helper functions
+    #####
+    def expand(self, *args, **kwargs):
+        return Equation(self.lhs.expand(*args, **kwargs), self.rhs.expand(
+            *args, **kwargs), check=False)
+
+    def simplify(self, *args, **kwargs):
+        return self._eval_simplify(*args, **kwargs)
+
+    def _eval_simplify(self, *args, **kwargs):
+        return Equation(self.lhs.simplify(*args, **kwargs), self.rhs.simplify(
+            *args, **kwargs), check=False)
+
+    def _eval_factor(self, *args, **kwargs):
+        # TODO: cancel out factors common to both sides.
+        return Equation(self.lhs.factor(*args, **kwargs), self.rhs.factor(
+            *args, **kwargs), check=False)
+
+    def factor(self, *args, **kwargs):
+        return self._eval_factor(*args, **kwargs)
+
+    def _eval_collect(self, *args, **kwargs):
+        from sympy.simplify.radsimp import collect
+        return Equation(collect(self.lhs, *args, **kwargs),
+                        collect(self.rhs, *args, **kwargs), check=False)
+
+    def collect(self, *args, **kwargs):
+        return self._eval_collect(*args, **kwargs)
+
+    def _eval_evalf(self, *args, **kwargs):
+        return Equation(self.lhs.evalf(*args, **kwargs),
+                        self.rhs.evalf(*args, **kwargs), check=False)
+
+    def evalf(self, *args, **kwargs):
+        return self._eval_evalf(*args, **kwargs)
+    n = evalf
```

Comment:
There should be a blank line between these two

Reply:
fixed in commit 6419fdad